migration-engine/connectors/sql-migration-connector/src/sql_renderer/sqlite_renderer.rs

```diff
@@ -77,6 +83,75 @@ impl SqlRenderer for SqliteFlavour {
     fn render_drop_enum(&self, _drop_enum: &crate::DropEnum) -> Vec<String> {
         Vec::new()
     }
+
+    fn render_redefine_tables(
+        &self,
+        tables: &[String],
+        differ: SqlSchemaDiffer<'_>,
+        database_info: &DatabaseInfo,
+    ) -> Vec<String> {
+        // Based on 'Making Other Kinds Of Table Schema Changes' from https://www.sqlite.org/lang_altertable.html
+        let mut result: Vec<String> = Vec::new();
+        let schema_name = database_info.connection_info().schema_name();
+        let sql_family = database_info.sql_family();
+
+        result.push("PRAGMA foreign_keys=OFF".to_string());
+
+        for table in tables {
+            let differ = TableDiffer {
+                database_info,
+                flavour: self,
+                previous: differ.previous.table_ref(table).expect(""),
+                next: differ.next.table_ref(table).expect(""),
+            };
+
+            let name_of_temporary_table = format!("new_{}", &differ.next.name());
+            let mut temporary_table = differ.next.table.clone();
+            temporary_table.name = name_of_temporary_table.clone();
+
+            // This is a hack, just to be able to render the CREATE TABLE.
+            let temporary_table = TableRef {
+                schema: differ.next.schema,
+                table: &temporary_table,
+            };
+
+            // TODO start transaction now. Unclear if we really want to do that.
```

Comment:
Seeing this comment now by chance: Right now i think we should actually start a transaction. With this new modeling `RedefineTables` is a sql migration step and therefore an atomic unit that is tracked by the applier. So it would make sense to start a transaction so that this appears as an atomic unit to the applier.
Too much for the scope of this PR but we could tackle that soonish.